### PR TITLE
Use 0.95 as default ratio for lucene radial search traversal similarity

### PR DESCRIPTION
**Pull request by junqiu-lei**
### Description
This PR set up Lucene engine traversalSimilarity value to 0.95 * resultSimilarity(min_score) based on benchmark result https://github.com/opensearch-project/k-NN/issues/814#issuecomment-2060195145 
 
### Issues Resolved
Part of https://github.com/opensearch-project/k-NN/issues/814
 
### Check List
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/knn/common/KNNConstants.java [hunk 1]
```diff
@@ -133,4 +133,6 @@ public class KNNConstants {
     // Please refer this github issue for more details for choosing this value:
     // https://github.com/opensearch-project/k-NN/issues/1049#issuecomment-1694741092
     public static int MAX_DISTANCE_COMPUTATIONS = 2048000;
+
+    public static float LUCENE_RADIAL_SEARCH_TRAVERSAL_SIMILARITY_RATIO = 0.95f;
```
Comment (ryanbogan) [inline src/main/java/org/opensearch/knn/common/KNNConstants.java:137]: Should we include default in the constant name somewhere?

**Review by ryanbogan — COMMENTED** [review]
One minor comment, but everything else looks good to me!

File: src/main/java/org/opensearch/knn/common/KNNConstants.java (hunk 1 above)
Comment (junqiu-lei) [reply src/main/java/org/opensearch/knn/common/KNNConstants.java]: Added

**Review by ryanbogan — DISMISSED** [review]
LGTM!

**Comment by navneet1v** [comment]
overall code looks good to me. Waiting for CIs to complete.

File: src/main/java/org/opensearch/knn/common/KNNConstants.java [hunk 2]
```diff
@@ -133,4 +133,6 @@ public class KNNConstants {
     // Please refer this github issue for more details for choosing this value:
     // https://github.com/opensearch-project/k-NN/issues/1049#issuecomment-1694741092
     public static int MAX_DISTANCE_COMPUTATIONS = 2048000;
+
+    public static float DEFAULT_LUCENE_RADIAL_SEARCH_TRAVERSAL_SIMILARITY_RATIO = 0.95f;
```
Comment (jmazanec15) [inline src/main/java/org/opensearch/knn/common/KNNConstants.java:137]: make final?

**Review by ryanbogan — APPROVED** [review]

**Comment by junqiu-lei** [comment]
Will do manually backport to 2.x with #1617

**Pull request merged**